### PR TITLE
[Console] Fix global console flag when used in chain

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #23876
| License       | MIT
| Doc PR        | 

Because SymfonyCon is great we can create pull request in it ! (this was preparer in the plane and I can push it just right now ;))

Finished in the #SymfonyConHackday2017

### CODE REVIEW

**Comment by chalasr** [comment]
For 2.7?

**Comment by Simperfit** [comment]
Yes I will rebase it :)
Le jeu. 16 nov. 2017 à 16:18, Robin Chalas <notifications@github.com> a
écrit :

File: src/Symfony/Component/Console/Input/ArgvInput.php [hunk 1]
```diff
@@ -168,12 +168,12 @@ private function parseArgument($token)
             $arg = $this->definition->getArgument($c);
             $this->arguments[$arg->getName()] = $arg->isArray() ? array($token) : $token;
 
-        // if last argument isArray(), append token to last argument
+            // if last argument isArray(), append token to last argument
```
Comment (ro0NL) [inline src/Symfony/Component/Console/Input/ArgvInput.php]: revert?

File: src/Symfony/Component/Console/Input/ArgvInput.php [hunk 2]
```diff
@@ -282,6 +282,14 @@ public function hasParameterOption($values)
                 if ($token === $value || 0 === strpos($token, $value.'=')) {
                     return true;
                 }
+
+                if (0 === strpos($token, '-') && 0 !== strpos($token, '--')) {
```
Comment (ro0NL) [inline src/Symfony/Component/Console/Input/ArgvInput.php]: i dont follow it really, but it works :+1: 

```
$ bin/console router:match -hq

before: exception (missing argument)
after: quiet :)

**Review by ro0NL — APPROVED** [review]

File: src/Symfony/Component/Console/Input/ArgvInput.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Console/Input/ArgvInput.php]: yes please

File: src/Symfony/Component/Console/Input/ArgvInput.php [hunk 3]
```diff
@@ -168,12 +168,12 @@ private function parseArgument($token)
...
             $this->arguments[$arg->getName()] = $arg->isArray() ? array($token) : $token;
 
-        // if last argument isArray(), append token to last argument
+            // if last argument isArray(), append token to last argument
         } elseif ($this->definition->hasArgument($c - 1) && $this->definition->getArgument($c - 1)->isArray()) {
             $arg = $this->definition->getArgument($c - 1);
             $this->arguments[$arg->getName()][] = $token;
 
-        // unexpected argument
+            // unexpected argument
```
Comment (fabpot) [inline src/Symfony/Component/Console/Input/ArgvInput.php]: same here

**Review by chalasr — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by chalasr** [comment]
Thanks for fixing this bug Hamza.

**Comment by Zwartpet** [comment]
@fabpot This results in a bug in v2.8.32. When I try to run a command the command is not actually executed but the Symfony shell is opened (same result as with adding the --shell option).
Did not check other releases though.

**Comment by Simperfit** [comment]
I will try to reproduce then fix this behaviour. Could you try to provide a
reproducer ?
Le mer. 13 déc. 2017 à 15:40, John Zwarthoed <notifications@github.com> a
écrit :

**Comment by Zwartpet** [comment]
Just running app/console already causes the issue for me (with symfony/console 2.8.32)

**Comment by Simperfit** [comment]
I've just tested with symfony/console 2.8.32 and I can't reproduce the problem : 

```
  - Updating symfony/symfony (v2.8.11 => v2.8.32): Loading from cache
  - Updating sensio/generator-bundle (v3.1.6 => v3.1.7): Loading from cache
  - Updating symfony/phpunit-bridge (v2.8.31 => v2.8.32): Loading from cache
```
```
#( 12/13/17@ 6:34PM )( hamza@MacBook-Pro-de-Amrouche ):~/projet/contrib/symfony/symfony-standard@reproduce-the-process✗✗✗
   app/console debug:config        

 // Provide the name of a bundle as the first argument of this command to dump its configuration.                       


Available registered bundles with their extension alias if available:
+----------------------------+------------------------+
| Bundle name                | Extension alias        |
+----------------------------+------------------------+
| AppBundle                  |                        |
| DebugBundle                | debug                  |
| DoctrineBundle             | doctrine               |
| FrameworkBundle            | framework              |
| MonologBundle              | monolog                |
| SecurityBundle             | security               |
| SensioDistributionBundle   | sensio_distribution    |
| SensioFrameworkExtraBundle | sensio_framework_extra |
| SensioGeneratorBundle      |                        |
| SwiftmailerBundle          | swiftmailer            |
| TwigBundle                 | twig                   |
| WebProfilerBundle          | web_profiler           |
+----------------------------+------------------------+
```
```
$    app/console debug:config --shell

      _____                  __
     / ____|                / _|
    | (___  _   _ _ __ ___ | |_ ___  _ __  _   _
     \___ \| | | | '_ ` _ \|  _/ _ \| '_ \| | | |
     ____) | |_| | | | | | | || (_) | | | | |_| |
    |_____/ \__, |_| |_| |_|_| \___/|_| |_|\__, |
             __/ |                          __/ |
            |___/                          |___/


Welcome to the Symfony shell (2.8.32 - app/dev/debug).

At the prompt, type help for some help,
or list to get a list of available commands.

To exit the shell, type ^D.

Symfony > 

```

**Comment by Zwartpet** [comment]
Welcome to the Symfony shell (2.8.32 - app/dev/debug).

At the prompt, type help for some help,
or list to get a list of available commands.

To exit the shell, type ^D.

Symfony >

Thats what i mean. You’re not supposed to be stuck in de symfony shell after running a command if i’m not mistaken.

**Comment by Zwartpet** [comment]
Oh wait, thats with —shell. I’ll check when i’m home

**Comment by Zwartpet** [comment]
Apparently it only occurs when running with 'test' in the environment.
For example app/console -e=docker would go fine, but app/console -e=docker_test would result in the shell. Even though the configurations are exactly the same.

**Comment by Zwartpet** [comment]
I am able to reproduce it on a fresh project:
```
$ symfony new my_project_name 2.8
$ cd my_project_name
$ app/console -e=test
```
Not able to reproduce in 3.4

**Comment by Simperfit** [comment]
Ok I understand it, I submit a PR to fix the bug

**Comment by Simperfit** [comment]
@Zwartpet see #25487

**Comment by nicolas-grekas** [comment]
breaks in #25825 apparently

**Comment by fabpot** [comment]
reverted

**Pull request merged**